asmrun/startup.c

```diff
@@ -121,6 +125,10 @@ value caml_startup_common(char_os **argv, int pooling)
   if (!caml_startup_aux(pooling))
     return Val_unit;
 
+#if defined(_WIN32) && WINDOWS_UNICODE
+  SetConsoleOutputCP(CP_UTF8);
```

Comment:
It would be worth a comment noting that this affects all output functions (i.e. both stdout and stderr), I think.

Two possible ideas:
1. Selecting CP_UTF8 only works if a TrueType font is selected (legacy Raster Fonts won't work, which is still the default case pre Windows 8, IIRC). You can detect this using this function

```c
int console_supports_unicode (void) {
  CONSOLE_FONT_INFOEX fontInfo;
  fontInfo.cbSize = sizeof(fontInfo);
  if (GetCurrentConsoleFontEx(GetStdHandle(STD_OUTPUT_HANDLE), FALSE, &fontInfo) && !(fontInfo.FontFamily & TMPF_TRUETYPE)) {
    return (GetCurrentConsoleFontEx(GetStdHandle(STD_ERROR_HANDLE), FALSE, &fontInfo) && (fontInfo.FontFamily & TMPF_TRUETYPE));
  } else {
    return 1;
  }
}
```
2. Having done that, it might be worth the addition of a function in `Sys` which exposes this (a bit like `Unix.has_symlink`)